internal/trace/metrics.go

```diff
@@ -14,67 +14,63 @@ var (
 	keyDialerID, _ = tag.NewKey("cloudsql_dialer_id")
 )
 
-var (
-	mLatencyMS = stats.Int64(
-		"/cloudsqlconn/latency",
-		"The latency in milliseconds per Dial",
-		stats.UnitMilliseconds,
-	)
-	latencyView = &view.View{
+// NewMetricsCollector registers all views. Without registering views, metrics will not
+// be reported. If any names of the registered views conflict, this function
+// returns an error to indicate a configuration problem.
+func NewMetricsCollector() (*MetricsCollector, error) {
+	mc := &MetricsCollector{
+		mLatencyMS: stats.Int64(
+			"/cloudsqlconn/latency",
+			"The latency in milliseconds per Dial",
+			stats.UnitMilliseconds,
+		),
+		mConnections: stats.Int64(
+			"/cloudsqlconn/connection",
+			"A connect or disconnect event to Cloud SQL",
+			stats.UnitDimensionless,
+		),
+	}
+
+	latencyView := &view.View{
 		Name:        "/cloudsqlconn/dial_latency",
-		Measure:     mLatencyMS,
+		Measure:     mc.mLatencyMS,
 		Description: "The distribution of dialer latencies (ms)",
 		// Latency in buckets, e.g., >=0ms, >=100ms, etc.
 		Aggregation: view.Distribution(0, 5, 25, 100, 250, 500, 1000, 2000, 5000, 30000),
 		TagKeys:     []tag.Key{keyInstance, keyDialerID},
 	}
-)
-
-var (
-	mConnections = stats.Int64(
-		"/cloudsqlconn/connection",
-		"A connect or disconnect event to Cloud SQL",
-		stats.UnitDimensionless,
-	)
-	connectionsView = &view.View{
+	connectionsView := &view.View{
 		Name:        "/cloudsqlconn/open_connections",
-		Measure:     mConnections,
-		Description: "The sum of Cloud SQL connections",
-		Aggregation: view.Sum(),
+		Measure:     mc.mConnections,
+		Description: "The current number of open Cloud SQL connections",
+		Aggregation: view.LastValue(),
```

Comment:
Why are we using LastValue as opposed to Count?

How does this effect things when there are multiple instances of the dialer or the dialer ends unexpectedly. Does it ever return to 0?

Reply:
At least for Cloud Monitoring, the LastValue type is the only type that results in a Gauge metric. Everything else, include Count, will be a Cumulative metric.

See https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/blob/dfce60d8dc4099a38012eb860ffba13f0e52e1a8/stackdriver.go#L33-L34.

I'll test to see what Cloud Monitoring does when a client doesn't report a zero value.

Reply:
What happens when a connection stays open for say 8 hours? What happens with the metric?

Reply:
When a dialer ends unexpectedly, the gauge metric does not return to zero. I think that's reasonable behavior. For metrics that might have a significant amount of time between them, the [Data Retention policy](https://cloud.google.com/monitoring/quotas#data_retention_policy) for custom metrics is 24 months. So I think the scenario for longer lived connections is well supported.

Reply:
Does the datapoint show up if it's outside of the graph window in pantheon? If I set the window to 1h, but the value was lated updated 8 hours previously does it still show up?

Reply:
@enocom this would be good to know but not necessarily blocking

Reply:
So I established two connections and let them sit for eight hours. When I set the Cloud Monitoring window to after the first point, I do see the metric appearing, which implies that the Stackdriver exporter is sending points repeatedly, even though they're not changing. Just when my program was about to close the connection, the TCP socket had been closed, so the value did not return to zero. Nonetheless, I think we're good here.